components/scripts/lib/info.sh

```diff
@@ -137,11 +137,42 @@ print_performance_metrics() {
   true
 }
 
-print_build_caching_leverage_metrics() {
+print_performance_characteristics() {
+  print_performance_characteristics_header
+
+  print_realized_build_time_savings
+
+  print_build_caching_leverage_metrics
+
+  print_executed_cacheable_tasks_warning
+}
+
+print_performance_characteristics_header() {
   print_bl
-  info "Build Caching Leverage"
+  info "Performance Characteristics"
   info "----------------------"
+}
+
+print_realized_build_time_savings() {
+  # Do not print realized build time savings at all if these values do not exist
+  # This can happen since build-scan-support-tool does not yet support these fields
+  if [[ -n "${effective_task_execution_duration[0]}" && -n "${effective_task_execution_duration[1]}" ]]; then
+    local value
+    value=""
+    # Only calculate realized build time savings when these values are non-zero
+    # These values can be returned as zero when an error occurs processing the Build Scan data
```

Comment:
-1: Can these values be zero if the project is fully unoptimized and has no build time savings (even if there is no error processing the Build Scan data)?

Reply:
As part of the contract changes in #283 I changed it so that these values can be returned from the CLI tools as empty (`""`) in case of an error. This allows us to differentiate between 0 values and errors. I think that should address your concern here.

Reply:
Yup, agree! 👍